src/main.ts

```diff
@@ -0,0 +1,137 @@
+import {NotificationMessage} from './NotificationMessage';
+import {EventType} from "./enums";
+import {schemaOnIssue, schemaOnIssueComment, schemaOnPullRequest, schemaOnPush} from "./eventPayloadSchemaBuilder";
+
+const core = require ('@actions/core');
+const path = require('path');
+const https = require('https');
+
+const jsonPath = path.join(__dirname, '..', 'resources', 'notification.json');
+const workflow = process.env.GITHUB_WORKFLOW;
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const github_event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+const trigger_event_name = process.env.GITHUB_EVENT_NAME;
+
+const msteams_webhook_url = core.getInput('webhook_url');
+const job_status = core.getInput('job_status');
+const event_id = core.getInput('event_id');
+
+export const parsedNotificationMessage = async function parse_event_to_message(event_payload_text: string): Promise<NotificationMessage> {
+    let details;
```

Comment:
I don't see any reason to have all these local bindings. It would be less mental overhead to just return a NotificationMessage instance from within each branch of the switch.

Reply:
If I understand you correctly, you would like to see something like that (an example from one case)
```
  case EventType.ISSUE_COMMENT:
      await schemaOnIssueComment.validate(eventPayloadText);
      parsedSchema = schemaOnIssueComment.cast(eventPayloadText);
      notificationMessage = new NotificationMessage(
          `**A Github issue comment was posted** by ${parsedSchema.sender.login} with content ${parsedSchema.comment.body}`,
           parsedSchema.issue.html_url,
           `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`);
     break;
```
Honestly, since we are building the returned strings based on each case, I do not see how this is more readable than actually having placeholder variables with the corresponding name of what is being set. Therefore unless you feel very strongly about this I would prefer not making this change.

Reply:
I am envisaging something like this: 

```
case EventType.PUSH:
                const parsed = await schemaOnpush.validate(eventPayloadText);
                const account = parsed.pusher.name;
                return {
                    message: `**Commit to GitHub** by ${account}`;
                    url: parsed.compare;
                    details: `Comment: ${parsed.head_commit.message}`;
                }; 
```
Key advantages (as I see it): validate _returns_ parsed data so no need to cast. All bindings are local to the case statement so I _know_ I don't need to look anywhere else to see what is going on - less mental overhead. There is no NotificationMessage class (and I lose _nothing_ as a result). I can see the return statement so I _know_ that nothing further down the function is relevant - no need to look or think, again, less mental overhead. 

So I have a few reasons there. These are not _necessary_ changes, but I think there is more to it than straight readability. 

If you are still unpersuaded, then that is also absolutely fine - just something to think about :)

Reply:
Following your vision and my thoughts on extensibility (most probably we will support other event types as well, what if we need some additional strings or attributes or what if we want to return more details in the notification message?) I used this return structure but as an interface, which allows me ensuring a specific contract once again. This is also aligned to your comment: https://github.com/efcloud/el-msteams-post-action/pull/1#discussion_r384998826

Reply:
Ah I think I see your concern. Returning the result as an object literal has no detrimental effect on type safety because typescript using structural typing. As long as the object literal returned from the function is _structurally_ compatible with the NotificationMessage type given in the function signature, typescript will be happy.

Reply:
Well you put it much much nicer than I did but you got me, thanks!